tools/xcode/templates/SFML/SFML CLT.xctemplate/main.cpp

```diff
@@ -65,13 +65,13 @@ int main()
         for (sf::Event event; window.pollEvent(event);)
         {
             // Close window: exit
-            if (event.type == sf::Event::Closed)
+            if (event.is<sf::Event::Closed>())
             {
                 window.close();
             }
 
             // Escape pressed: exit
-            if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Key::Escape)
+            if (event.is<sf::Event::KeyPressed>() && event.get<sf::Event::KeyPressed>().code == sf::Keyboard::Key::Escape)
```

Comment:
I suggest using `getIf` + C++17 `if` with initializer here.

Reply:
I was fine changing the first instance you pointed out but don't want to change them all.

Reply:
It's exactly the same code as above. HOw is the one above worth changing but this one isn't?

Reply:
I didn't think the first was worth changing per se. I thought you only wanted that pattern change in one place so I obliged rather than argue about it. However now that you want this pattern changed in many places I'm pushing back a bit harder.

Reply:
The reason why I believe the pattern I suggested is technically superior to the one you have here is that it prevents mismatches of the type `e.is<X>() && e.get<Y>()`. 

Even if a mismatch is not present, any reader of the source code needs to visually check if the two types are matching. The pattern I suggested prevents the mismatch from happening in the first place and prevents readers from having to double-check.

My recommendation is to apply that pattern wherever possible. 

If you do not agree with this recommendation, please explain why you believe that the current pattern is superior to the one I am suggesting, given my technical argumentations above.

If you still do not agree with the recommendation even after that, please revert the place that you have changed above to the pattern I consider inferior to at least maintain consistency in the codebase.